client/src/app/components/gui-editor/sidebar/editor-sidebar.component.ts

```diff
@@ -0,0 +1,80 @@
+import {
+    Component,
+    style,
+    animate,
+    state,
+    transition,
+    trigger,
+    EventEmitter,
+    Output
+} from "@angular/core";
+import {NgFor} from "@angular/common";
+import {GuiEditorService} from "../gui-editor.service";
+import {VisibilityState} from "../animation.states";
+import {GuiEditorEvent, GuiEditorEventType, ShowSidebarEvent} from "../gui-editor.events";
+
+require ("./editor-sidebar.component.scss");
+
+export enum SidebarType {
+    expression,
+    editor
+}
+
+/** TODO: make this switch between an expression editor and an object inspector*/
+@Component({
+    selector: "editor-sidebar",
+    directives: [NgFor],
+    animations: [
+        trigger("sidebarState", [
+            state("visible", style({
+                width: '40%',
+                display: 'block',
+                overflowY: "auto"
+            })),
+            state("hidden", style({
+                width: '10%',
+                display: 'none',
+                overflowY: "hidden"
+            })),
+            transition("hidden => visible", animate("100ms ease-in")),
+            transition("visible => hidden", animate("100ms ease-out"))
+        ])
+    ],
+    template: `
+                    <div class="rightSidebar" @sidebarState="sidebarState">
+                        <div class="collapseIcon">
+                            <i class="fa fa-lg fa-caret-left" (click)="collapseSidebar()"></i>
+                        </div>
+                        <div class="sideBarContent">
+                            <!-- TODO: add expression and objects editor -->
+                            This is the right sidebar content
+                        </div>
+                    </div>
+    `
+})
+export class EditorSidebarComponent {
+    /** Emit changes of the sidebar animation to the parent component */
+    @Output()
+    private sidebarVisibility = new EventEmitter();
```

Comment:
Seed simpler to listen to an event on the parent, instead of sending it through a service. I'm using this as directive, so I would not like to access the its class property directly. (If that is what you meant)